tests/container_checker/test_container_checker.py

```diff
@@ -0,0 +1,291 @@
+"""
+Test the feature of container_checker
+"""
+import logging
+
+import pytest
+
+from pkg_resources import parse_version
+from tests.common import config_reload
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.helpers.assertions import pytest_require
+from tests.common.helpers.dut_utils import check_container_state
+from tests.common.helpers.dut_utils import clear_failed_flag_and_restart
+from tests.common.helpers.dut_utils import is_hitting_start_limit
+from tests.common.helpers.dut_utils import is_container_running
+from tests.common.plugins.loganalyzer.loganalyzer import LogAnalyzer, LogAnalyzerError
+from tests.common.utilities import wait_until
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('any'),
+    pytest.mark.disable_loganalyzer
+]
+
+CONTAINER_CHECK_INTERVAL_SECS = 1
+CONTAINER_STOP_THRESHOLD_SECS = 30
+CONTAINER_RESTART_THRESHOLD_SECS = 180
+
+
+@pytest.fixture(autouse=True, scope='module')
+def config_reload_after_tests(duthost):
+    yield
+    config_reload(duthost)
+
+
+@pytest.fixture(autouse=True, scope="module")
+def check_image_version(duthost):
+    """Skips this test if the SONiC image installed on DUT was 201911 or old version.
+
+    Args:
+        duthost: Host DUT.
+
+    Return:
+        None.
+    """
+    pytest_require(parse_version(duthost.kernel_version) > parse_version("4.9.0"),
+                   "Test was not supported for 201911 and older image version!")
+
+
+@pytest.fixture(autouse=True, scope="module")
+def update_monit_service(duthost):
+    """Update Monit configuration and restart it.
+
+    This function will first reduce the monitoring interval of container checker
+    from 5 minutes to 2 minutes, then restart Monit service without delaying. After
+    testing, these two changes will be rolled back.
+
+    Args:
+        duthost: name of Host DUT.
+
+    Return:
+        None.
+    """
+    logger.info("Back up Monit configuration files.")
+    duthost.shell("sudo cp /etc/monit/monitrc /tmp/")
+    duthost.shell("sudo cp /etc/monit/conf.d/sonic-host /tmp/")
```

Comment:
Suggest using `cp -f` here, just in case there are stale files left over.

Reply:
Great suggestion! Updated.